### PR TITLE
Added some updates to the README

### PR DESCRIPTION
**Pull request by nlordell**
When getting started with the dex-services I ran into some issues when following the README instructions.
- The NodeJS version matters because some dependencies fail to build under more recent NodeJS versions
- Not doing an `npm install` with the `--production` flag causes docker build issues since the git directory is not being copied into the container and NPM shells out to git during the install process which fails because of this.
- Added some arbitrary known good versions for other tools

Note that the docker build for truffle container is still failing because of another unrelated issue.

Test plan:
Go through the getting started section on a fresh clone of the dex-services

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,15 +39,25 @@ Thus, our data layer has to rely only on the data emitted by the EVM. It cannot
 
 More components, e.g. a watchtower to challenge invalid state transitions, will be added in the future.
 
-## Installation
+## Getting Started
+
+### Requirements
+
+- Rust ^1.37
+- NodeJS ^11.0, starting with version 12 some deprecated APIs were removed that cause `scrypt`, `keccak`, `secp256k1`, and `sha3` packages to fail to build
```
Comment (bh2smith) [inline README.md]: Doesn't the superscript on `^11.0` include `12` as acceptable version? Or does the `^` only allow all `11.X.X` versions?

Comment (nlordell) [reply README.md]: `^x.y.z <=> >=x.y.z && <(x+1).0.0`

**Comment by nlordell** [comment]
Sounds good, I'll remove the `--production` flag from the README (which makes sense in our case because we are building `dex-contracts`)

**Comment by nlordell** [comment]
Hey, master has been updated, should I:
- merge `origin/master` into `readme_updates`
- rebase `readme_updates` onto `origin/master` (required force push of `readme_updates` branch)

Not sure what you guys usually do.

**Comment by nlordell** [comment]
Ended up merging since, as @bh2smith pointed out to me we squash all the commits on merge so it doesn't really matter.

**Comment by fleupold** [comment]
Yes that works. I personally tend to rebase and force push to the PR branch again (this way at least for the purpose of review my commit history is preserved and can be easily applied to other branches). But since it all gets squashed down eventually it doesn't matter

**Pull request merged**